packages/fluentui/react-bindings/src/hooks/useSlotProps.ts

```diff
@@ -0,0 +1,20 @@
+type UseSlotPropsResult = (slot: string) => Record<string, any>;
+
+const useSlotProps = <P = {}>(
+  slotPropsChain: Record<string, (props: P) => Record<string, any> | Record<string, any>>[],
+  props: P,
+): UseSlotPropsResult => {
+  const getSlotProps: UseSlotPropsResult = slot => {
+    return slotPropsChain.reduce(
+      (acc, slotProps) => ({
+        ...acc,
+        ...(typeof slotProps[slot] === 'function' ? slotProps[slot](props) : slotProps[slot] || {}),
```

Comment:
Why this is not a part of `compose`?

Reply:
I mean the whole function

Reply:
Do you mean we should have as part of `composeOption` the `getSlotProps` function, so we can just use `composeOptions.getSlotProps('content')` for example?

Reply:
Yep. Or even `composeOptions.slotProps.content` as I don't understand why we need `getSlotProps()` as function: to component we pass resolved options already

Reply:
We do not have the props inside compose, that is the reason why I initially made it a separate hook. If we return some function from compose, I think we will need to still invoke it with the properties inside the component. Am I missing something?

Reply:
If we have `mapPropsToSlots` as a compose thingy, we will definitely need a hook in order to resolve them, again because we need the props, similarly as `useStyles` uses for now `unstable_props`

Reply:
But we have them all there 🤔

https://github.com/microsoft/fluentui/blob/a4a37e45bee51f66b5c8cabd87590df6e43c6e87/packages/react-compose/src/compose.ts#L17

Reply:
So after some thoughts.. What I have is following identically what we do with `mapPropsToStylesProps` - compose is creating chain, and than the hook is invoking everything with the props. We don't have access to the props inside compose.

Reply:
It's a different as in `useStyles` we are using `mapPropsToStyles` in `useStyles` options. But here as functions in chain are available at `composeOptions`

Reply:
Ok let me try to see if everything will work as expected then... I am going to update `slotProps` to be `mapPropsToSlots: (props) => ({ content: {} })` and try to remove the `useSlotProps` hook

Reply:
Just to conclude, after offline discussions we decided to change API of compose to `mapPropsToSlots: (props) => ({ content: {} })`, but still use a hook for getting the slot props, as for some components we will need both props + state for resolving the slot props.

Reply:
My preference would be in `compose`: keep `slots` and `slotProps` as names. Either of these could be and static object map, or a function taking in `state` and returning the object map. It keeps the consumer experience super clean:

```tsx
compose(Button, {
  slots: {
     icon: MyContainer
  },
  slotProps: {
     icon: { data-is-foo: 'bar'
  }
}

We should consider moving `mapPropsToStyles` to something maybe higher level to be layered on demand: it is a css-in-js ism specifically for caching so as we move to static

I think a function like `mergeProps` would be responsible for deriving the final `slots` and `slotProps` to be distributed. It would need to be called after state has been reconciled. It could manage resolving things like slots/slotProps and classes functions which all should take in state to resolve their stuff.

This means that compose simply glues together the options and passes them along, while the mergeProps helper could finalize the results for consumption in the template.

Reply:
We already have `mapPropsToStyleProps`, that's why I decided in the end to go with `mapPropsToSlotPros`. If we decide to change the API of the compose we can rename those easily to just stylesProps and slotProps, but for now I would rather keep the names close...